src/applications/personalization/profile-2/components/connected-apps/actions/index.js

```diff
@@ -49,7 +49,7 @@ export function deleteConnectedApp(appId) {
       return;
     }
 
-    apiRequest(`${grantsUrl}/${appId}`, { method: 'DELETE' })
+    await apiRequest(`${grantsUrl}/${appId}`, { method: 'DELETE' })
```

Comment:
Needed to add `await` here to make sure the test waits for the 2nd time `dispatch` is called: https://github.com/department-of-veterans-affairs/vets-website/pull/13023/files#diff-006ba99817a84f8cd2cdc579c0383f8fR41